File: app/src/main/java/com/harishtk/goldrate/app/thread/BackgroundThreadFactory.java
package com.harishtk.goldrate.app.thread;

import android.os.Process;
import android.util.Log;

import java.util.Locale;
import java.util.concurrent.ThreadFactory;

public class BackgroundThreadFactory implements ThreadFactory {
    public static final String TAG = BackgroundThreadFactory.class.getSimpleName();
    public static final int sTag = 1;

    @Override
    public Thread newThread(Runnable r) {
        Thread thread = new Thread(r);
        thread.setName("GoldSpider" + sTag);
        thread.setPriority(Process.THREAD_PRIORITY_BACKGROUND);

        thread.setUncaughtExceptionHandler((t, e) ->
                Log.e(TAG, String.format(Locale.ENGLISH, "%s encountered an error: %s", thread.getName(), e.getMessage())));

        return thread;
    }
}

File: app/src/main/java/com/harishtk/goldrate/app/thread/BackgroundThreadFactory.kt
package com.harishtk.goldrate.app.thread

import android.os.Process
import android.util.Log
import java.util.*
import java.util.concurrent.ThreadFactory

class BackgroundThreadFactory: ThreadFactory {

    override fun newThread(r: Runnable?): Thread {
        return Thread(r).apply {
            name = "GoldSpider$sTag"
            priority = Process.THREAD_PRIORITY_BACKGROUND
            uncaughtExceptionHandler = Thread.UncaughtExceptionHandler { _: Thread?, e: Throwable -> Log.e(TAG, String.format(Locale.ENGLISH, "%s encountered an error: %s", name, e.message)) }
        }
    }

    companion object {
        val TAG: String = BackgroundThreadFactory::class.java.simpleName
        const val sTag: Int = 1
    }

}